[heyitssakinah] iP

heyitssakinah opened the pull request:
# Abu Hurairah Bot 
> Designed for muggers by muggers

## 💅 Description
Abu Hurairah was designed as an **efficient day-to-day** life organiser allowing muggers to keep track of <ins>tasks, events, deadlines</ins>, and more

## 🥵 Features 

- Task Management
- Deadline Retrieval
- Event Scheduling

### Upcoming Specials 

- [X] Changing Deadlines 
- [ ] Tagging Tasks (Importance levels)


## 🦾 How To Use

1. Download the [JAR file here](https://github.com/heyitssakinah/ip/releases/tag/A-Jar)
2. Run it on your local computer `java -jar shin.jar`
3. Click [this link](https://www.youtube.com/watch?v=xvFZjo5PgG0)
4. Have at it!

### The main method

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

heyitssakinah committed:
Named my bot

heyitssakinah committed:
trivial echo inputs

heyitssakinah committed:
Add, List

heyitssakinah committed:
Update functionality Mark as Done

heyitssakinah committed:
Created Task class and subclasses

heyitssakinah committed:
Add error handling

heyitssakinah committed:
Modified error handling

heyitssakinah committed:
Add delete function

heyitssakinah committed:
log ENUMS

heyitssakinah committed:
Add save data on hard-disk

heyitssakinah committed:
Test pushing to branch

heyitssakinah committed:
Add date and times to event tasks

heyitssakinah committed:
Add OOP files

heyitssakinah committed:
Create package

heyitssakinah committed:
Merge branch 'branch-A-Gradle'

heyitssakinah committed:
Add gradle

heyitssakinah committed:
Add JUnit tests

heyitssakinah committed:
Add javadocs

heyitssakinah committed:
Tweak to fit coding standard

heyitssakinah committed:
Add find function

heyitssakinah committed:
Merge branch 'branch-CodingStandard'

heyitssakinah committed:
Fix merge conflicts

kaungzinye commented on src/main/java/abuhurairah/TaskList.java:
```diff
@@ -0,0 +1,217 @@
+                                .filter(overdueTask -> overdueTask.isOverdue() && !overdueTask.isComplete())
+                                .forEach(overdueTask -> ui.serve("" + overdueTask.toString(), isPrintable));
+                        ui.separator();
+                    }
```
Ensure that the indentation level of the case is the same as the indentation of the switch statement in accordance to the Java Coding Standards given in the module. Other than that, very nicely structured code!

kaungzinye reviewed (commented):
LGTM! Just some nits and formatting to fix

heyitssakinah committed:
Add Checkstyle

zwliew commented on src/main/java/abuhurairah/AbuHurairah.java:
```diff
@@ -0,0 +1,47 @@
+        Parser parser;
+        String name = "Abu Hurairah";
+
+        storage = new Storage("./AbuHurairahHistory.txt");
+        parser = new Parser();
```
It would be more readable to initialize `storage` and `parser` in the same lines where you declared them.

zwliew commented on src/main/java/abuhurairah/Event.java:
```diff
@@ -0,0 +1,35 @@
+     */
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + "(from:" + this.start + "to:" + this.end + ")";
```
Consider using the `String.format` function to make string formatting easier and more readable.

zwliew commented on src/main/java/abuhurairah/Storage.java:
```diff
@@ -0,0 +1,85 @@
+            // Start a new chat Bot if file does not exist
+            ui.showWelcome(name);
+        }
+        return 0;
```
In the case where `FileNotFoundException` was previously caught, perhaps the return value of this function should be something else, like an error value? I think an invalid return value like `-1` would be fitting, but maybe an `Optional<Integer>` would help to denote error cases like this in a more type-safe way.

zwliew reviewed (commented):
Looks good overall! I have some minor comments

heyitssakinah committed:
Add javaFX

heyitssakinah committed:
Modify structure

heyitssakinah committed:
TaskTracker class: create to track completed and remaining tasks

Required feat missing.

Allows the user to see how many tasks they have left.

Let's abstract out getRemainingTasks method and keep track of tasks in a TaskTracker instance.

heyitssakinah committed:
add assertions for data integrity and validation

Code lacks sufficient validation checks, which could lead to unexpected behavior or errors during runtime.

Let's add assertions to enforce preconditions and postconditions in storage operations and task management.

Assertions are used because they are lightweight and efficient fo
debugging, allowing potential issues to be caught during development.

heyitssakinah committed:
Refactor code into packages

The project currently has a single directory structure.

To improve organization, modularity, and maintainability, the code will be refactored into distinct packages. This will group related classes together and improve the structure of the project.

Let's create the command, storage, ui, and task packages and move the related files into these packages.

This separation will lay the foundation for a cleaner, more modular codebase that is easier to navigate and extend.

heyitssakinah committed:
Add java docs

heyitssakinah committed:
Fix codequality: checkstyle

heyitssakinah committed:
Fix tests

heyitssakinah committed:
Add Streams to Retrieve commands

heyitssakinah committed:
Merge pull request #2 from heyitssakinah/branch-A-Assertions

add assertions for data integrity and validation

heyitssakinah committed:
Merge pull request #3 from heyitssakinah/branch-A-Streams

Add Streams to Retrieve commands

heyitssakinah committed:
Fix merge conflicts

heyitssakinah committed:
Merge pull request #4 from heyitssakinah/branch-A-CodeQuality

Branch a code quality

heyitssakinah committed:
Feat Add C-Friendlier

heyitssakinah committed:
Add styles.css

heyitssakinah committed:
Add personalised messgaes

heyitssakinah committed:
Add resources

heyitssakinah committed:
Merge branch 'branch-A-Personality'

heyitssakinah committed:
Add custom Exception

heyitssakinah committed:
Add new tests

heyitssakinah committed:
Add java docs in task tracker using AI

heyitssakinah committed:
Update using Checktyle

heyitssakinah committed:
Add Ui

heyitssakinah committed:
Update README.md

heyitssakinah committed:
Update README.md

heyitssakinah committed:
Update README.md

heyitssakinah committed:
Update README.md

heyitssakinah committed:
Update files for checkstyle and dependencies

heyitssakinah committed:
Update README.md

heyitssakinah committed:
Add exception message for command type

heyitssakinah committed:
Update README.md
